extension/lib/shopify/storefront-client/RequestFailedError.js

```diff
@@ -0,0 +1,23 @@
+class ShopifyStorefrontClientRequestFailedError extends Error {
+  constructor (statusCode, url, requestBody) {
+    super(`Request failed on ${url}`)
+
+    this._statusCode = statusCode
+    this._url = url
+    this._requestBody = requestBody
+  }
+
+  get statusCode () {
```

Comment:
please add jsdoc for all method + constructor

Reply:
done